File: projects/grind-75/python/string_lc_questions/longest_palindrome.py
class Solution:
    @classmethod
    def longestPalindrome(self, s: str) -> int:
        s_count = {}
        for c in s:
            s_count[c] = s_count.get(c, 0) + 1
        res = 0
        has_odd_amount_of_characters = False
        for count in s_count.values():
            if count % 2 == 1:
                has_odd_amount_of_characters = True
                res += count - 1
            else:
                res += count
        return res + 1 if has_odd_amount_of_characters else res

File: projects/grind-75/typescript/string/longest-palindrome.ts
export default function longestPalindrome(s: string): number {
  const s_count: { [c: string]: number } = {};
  for (const c of s) {
    s_count[c] = (s_count[c] || 0) + 1;
  }
  let res = 0;
  let hasOddAmountOfCharacters = false;
  for (const count of Object.values(s_count)) {
    if (count % 2 === 1) {
      hasOddAmountOfCharacters = true;
      res += count - 1;
    } else {
      res += count;
    }
  }
  return hasOddAmountOfCharacters ? res + 1 : res;
}
